Packages/com.unity.inputsystem/InputSystem/Controls/InputControlPath.cs

```diff
@@ -481,6 +481,12 @@ private static string FindControlLayoutRecursive(ref PathParser parser, InputCon
             return currentResult;
         }
 
+        internal static bool BindingPathMatchesMask(string mask, string bindingPath)
+        {
+            var bindingPathString = new InternedString(bindingPath);
```

Comment:
I'm a little worried about the InternedString creation here which is creating a lowercase variant of the bindingPath string which will be doing a GC allocation 

This BindingPathMatchesMask is called inside a loop over all bindings so this will be generating a lot of garbage.

Perhaps StringMatches could be altered to take a string variant. However I think it uses InternedString for the speed of the internal checks (as we've seen in the separate PR about turkish letter support)

I'm also seeing that StringMatches says the second 'matchTo' parameter may not contain wildcards and I think the bindingPath is allowed to contain wildcards.

Reply:
Yes binding path can contain wildcards
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.5/manual/ActionBindings.html

> A Binding path can also contain wildcards, such as <Gamepad>/button*